### PR TITLE
1939: Fixing Preset Loading and Reworking Campaign Options Initial Value Load

### LINKED ISSUE #1939: 47.10-Dev Unexpected campaign options changes
### Environment
v47.10-Dev (downloaded on 8/17/2020)
Windows 10
Java version 1.8.0_261

### Description
First, a little context. This AtB campaign save originated in v47.7, was moved to v47.9, and was then further moved to the current dev-build in order to get update #1931 (having encountered severe issues with issue #1922). This fix seems to work quite well, and I have not encountered #1922 since. 

However, while things seemed to be working fine for several hours of use yesterday, today I have encountered an odd issue: campaign options changes that I didn't initiate. Finding/reproducing this one might be a pain, because I'm not sure exactly when the changes _occurred_. The first change (significant changes in the "Skills" tab of Campaign Options, altering the base Target Number for all Gunnery skills in addition to a variety of other parameters on that page) I only noticed after loading the campaign save today. I manually corrected this, and continued playing. The _second_ issue (changes in the XP costs for Special Pilot Abilities) may have occured at the same time, or possibly not; either way I did not notice it until significantly later, after finishing a Base Defense battle. I will say that the change in the "Skills" tab definitely did not occur immediately upon loading the campaign in MekHQ v47.10-Dev, because the change in the listed Gunnery values of units in the TO&E is distinctive, and I did not see that yesterday. Therefore, it is possible that this bug is intermittent. I apologize that I cannot provide more detailed reproduction steps. 

### Files
The below file is the last save of the campaign file from v47.9, to serve as a reference for Campaign Options. 
[Coppercoats.47.9.cpnx.zip](https://github.com/MegaMek/mekhq/files/5093748/Coppercoats.47.9.cpnx.zip)

The below file is the most recent save of the campaign file from v47.10-Dev. If you open both campaign files, you can see that the XP costs for Special Abilities are different in this file (or at least, that is how it shows up on my machine ...). I did _not_ initiate this change myself. There may be other changes; I have not exhaustively cross-checked the Campaign Options, or any other parameters. Unit and personnel details not being identical is expected, as this save is two months of in-game time ahead of the "baseline" save. 
[Coppercoats.zip](https://github.com/MegaMek/mekhq/files/5093753/Coppercoats.zip)

Logs are included, but I suspect they have nothing useful in them. 
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/5093727/mekhqlog.txt)
[megameklog.txt](https://github.com/MegaMek/mekhq/files/5093728/megameklog.txt)

No custom units or other files.

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #1939

### CODE REVIEW

File: MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java [hunk 1]
```diff
@@ -4609,13 +4354,434 @@ private void fillRankInfo() {
...
+        chkUseDropShips.setSelected(options.getUseDropShips());
+        chkUseWeatherConditions.setSelected(options.getUseWeatherConditions());
+        chkUseLightConditions.setSelected(options.getUseLightConditions());
+        chkUsePlanetaryConditions.setSelected(options.getUsePlanetaryConditions());
+        chkUseAtBCapture.setSelected(options.getUseAtBCapture());
+        spnStartGameDelay.setValue(options.getStartGameDelay());
+        //endregion Against the Bot Tab
+
+        //region Miscellaneous Tab
+        chkHistoricalDailyLog.setSelected(options.historicalDailyLog());
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java:4760]: This probably ought to go into MekHQ preferences, but not for this PR.

File: MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java [hunk 2]
```diff
@@ -4609,13 +4354,434 @@ private void fillRankInfo() {
...
+                array[i][j] = Integer.toString(type.getCost(j));
+            }
+            i++;
+        }
+        return array;
+    }
+    //endregion Initialization
+
+    @SuppressWarnings(value = "unused") // FIXME:
     private void tableRanksValueChanged(javax.swing.event.ListSelectionEvent evt) {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java:4779]: What's the point of having this method if it does nothing and isn't used anyway?

File: MekHQ/src/mekhq/campaign/GamePreset.java [hunk 3]
```diff
@@ -228,53 +234,41 @@ else if (xn.equalsIgnoreCase("campaignOptions")) {
...
+     */
+    public static List<GamePreset> getGamePresetsIn() {
+        List<GamePreset> presets = new ArrayList<>();
+
+        File[] files = Utilities.getAllFiles(MekHQ.PRESET_DIR, (dir, name) -> name.toLowerCase().endsWith(".xml"));
+        for (File file : files) {
+            // And then load the campaign object from it.
+            GamePreset preset;
+
+            try (FileInputStream fis = new FileInputStream(file)) {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/GamePreset.java:263]: I assume this construct automatically closes the input stream regardless of whether the try/catch block throws an exception or not?

**Review by NickAragua — COMMENTED** [review]
Looks fine, just a couple of questions.

File: MekHQ/src/mekhq/campaign/GamePreset.java (hunk 3 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/GamePreset.java:263]: Exactly, so you don't have to worry about adding it yourself or not having it closed following an exception.

File: MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java (hunk 2 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java:4779]: This is something that we should have, but was broken it seems.

File: MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java:4760]: On my current list to do, just not in this PR.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**